lib/oga/xml/node.rb

```diff
@@ -55,6 +55,21 @@ def children=(nodes)
         end
       end
 
+      # Adds a child (or creates a child text node) to this node.
+      #
+      # @param [Oga::XML::Node|String] child
+
+      def <<(child)
+        child = Text.new(:text => child) if child.is_a?(String)
+
+        unless child.is_a?(Node)
+          raise ArgumentError, 'Can only add nodes or text.'
+        end
+
+        children << child
+        self
+      end
+
```

Comment:
So, first, I think there are two separate issues:

- Should this be allowed for nodes?
- Should this be allowed for nodes *and* text?

I'll start by saying that I could argue against both of the above, but after thinking about it a bit, have a slight preference for keeping at least one of them.

For appending nodes to a parent, I'll note that all of `REXML` `Nokogiri` and `libxml-ruby` *do* support appending nodes to parents via the `<<` operator.

In practice, I don't think it's ambiguous what the operator does.  Reading code from the other three libs doesn't cause any cognitive friction for me in this particular case.

The second question is I feel is harder -- my inclusion of text insertion wasn't so much out of a sense that the `<<` operator was the right solution, but that the current method felt clunky for such a common operation.

Since I wrote the code out to compare, I thought I'd include that here:

    require 'rexml/document'
    puts 'rexml:'
    
    doc = REXML::Document.new
    foo = REXML::Element.new('foo')
    bar = REXML::Element.new('bar')
    doc << foo
    foo << bar
    bar.text = 'baz'
    
    puts doc.to_s
    
    require 'nokogiri'
    puts 'nokogiri:'
    
    doc = Nokogiri::XML::Document.new
    foo = Nokogiri::XML::Element.new('foo', doc)
    bar = Nokogiri::XML::Element.new('bar', doc)
    doc << foo
    foo << bar
    bar << 'baz'
    
    puts doc.to_s
    
    require 'libxml'
    puts 'libxml:'
    
    doc = LibXML::XML::Document.new
    foo = LibXML::XML::Node.new('foo')
    bar = LibXML::XML::Node.new('bar')
    doc.root = foo
    foo << bar
    bar << 'baz'
    
    puts doc.to_s
    
    require 'oga'
    puts 'oga:'
    
    doc = Oga::XML::Document.new
    foo = Oga::XML::Element.new(:name => 'foo')
    bar = Oga::XML::Element.new(:name => 'bar')
    doc.children << foo
    foo.children << bar
    bar.inner_text = 'baz'
    
    puts doc.to_xml